### PR TITLE
add base entities

### PR DESCRIPTION
**Pull request by andreyshspb**

### CODE REVIEW

File: src/adjustment.cpp [hunk 1]
```diff
@@ -26,26 +28,39 @@ namespace TrafficEngineering {
     void adjustmentDfs(int curSwitchId, const Tunnel& tunnel, const Network& network) {
         maxLatency[curSwitchId] = 0;
         minLatency[curSwitchId] = SIZE_MAX;
```
Comment (pschuprikov) [inline src/adjustment.cpp:30]: Compile error on my machine: `SIZE_MAX` undefined. Maybe use `std::numeric_limits<size_t>::max()`? or better introduce `using Latency = size_t;` and then `std::numeric_limits<Latency>::max()`

File: src/adjustment.cpp [hunk 2]
```diff
@@ -26,26 +28,39 @@ namespace TrafficEngineering {
...
+            // useful variables
+            size_t transfer_time = tunnel.getPacketSize() / network.getBandwidth(curSwitchId, next);
+
+            // maximum latency relaxing
             size_t curMaxLatency = 0;
             curMaxLatency += maxLatency[next];
             curMaxLatency += network.getMinDelay(curSwitchId, next);
             curMaxLatency += network.getMaxJitter(curSwitchId, next);
             curMaxLatency += summaryTime[{curSwitchId, next}];
             maxLatency[curSwitchId] = std::max(maxLatency[curSwitchId], curMaxLatency);
```
Comment (pschuprikov) [inline src/adjustment.cpp]: extract to `relaxMaximumLatency(..)` instead of adding a comment

File: src/adjustment.cpp [hunk 3]
```diff
@@ -26,26 +28,39 @@ namespace TrafficEngineering {
     void adjustmentDfs(int curSwitchId, const Tunnel& tunnel, const Network& network) {
         maxLatency[curSwitchId] = 0;
         minLatency[curSwitchId] = SIZE_MAX;
+        periodLowerBound[curSwitchId] = 0;
         for (auto next : tunnel.getNextSwitchIds(curSwitchId)) {
             adjustmentDfs(next, tunnel, network);
 
+            // useful variables
+            size_t transfer_time = tunnel.getPacketSize() / network.getBandwidth(curSwitchId, next);
```
Comment (pschuprikov) [inline src/adjustment.cpp:36]: can it be made `const` and `auto`?

File: src/adjustment.cpp [hunk 4]
```diff
@@ -1,26 +1,66 @@
...
+#include <map>
+
 
 namespace TrafficEngineering {
 
-    std::vector<size_t> min_latency;
-    std::vector<size_t> max_latency;
+    std::map<int, size_t> minLatency;
+    std::map<int, size_t> maxLatency;
+    std::map<std::pair<int, int>, size_t> summaryTime;
```
Comment (pschuprikov) [inline src/adjustment.cpp:12]: `std::pair<int,int>` should probably be a separate type. e.g. `struct Endpoints { const int src; const int dst; }`, or even add `using NodeID = int` and replace `int`s with `NodeID`s.

File: src/adjustment.cpp (hunk 1 above)
Comment (andreyshspb) [reply src/adjustment.cpp:30]: Should I do the same for jitter?

**Pull request merged**